docs/tags.rst

```diff
@@ -92,11 +93,102 @@ Reference
+
+    :param tuple python_version: A two-item tuple representing the compatible
+                                 version of Python. Defaults to
+                                 ``sys.version_info[:2]``.
+    :param str interpreter: The name of the interpreter (if known).
+    :param Iterable platforms: Iterable of compatible platforms. Defaults to the
+                               platforms compatible with the current system.
+
+.. function:: cpython_tags(python_version=sys.version_info[:2], abis=None, platforms=None, *, warn=False)
+
+    Yields the tags for the CPython interpreter.
+
+    The specific tags generated are:
+
+    - ``cp<python_version>-<abi>-<platform>``
+    - ``cp<python_version>-abi3-<platform>``
+    - ``cp<python_version>-none-<platform>``
+    - ``cp<older version>-abi3-<platform>`` where "older version" is all older
+      minor versions down to Python 3.2 (when ``abi3`` was introduced)
+
+    :param tuple python_version: A tuple representing the targetted Python
+                                 version.
+    :param Iterable abis: Iterable of compatible ABIs. Defaults to the ABIs
+                          compatible with the current system.
+    :param Iterable platforms: Iterable of compatible platforms. Defaults to the
+                               platforms compatible with the current system.
+    :param bool warn: Whether warnings should be logged. Defaults to ``False``.
+
+.. function:: pypy_tags(interpreter=None, abis=None, platforms=None)
+
+    Yields the tags for the PyPy intrepreter.
+
+    The specific tags are those returned by :func:`generic_tags`, but with
+    *interpreter* inferred for the current system if not provided.
+
+    :param str interpreter: The name of the interpreter. Defaults the current
+                            PyPy interpreter.
+    :param Iterable abis: Iterable of compatible ABIs. Defaults to the ABIs
+                          compatible with the current system.
+    :param Iterable platforms: Iterable of compatible platforms. Defaults to the
+                               platforms compatible with the current system.
+
+.. function:: generic_tags(interpreter=None, abis=None, platforms=None, *, warn=False)
+
+    Yields the tags for an interpreter in a non-specialized fashion.
+
+    This function should be used if one of the other interpreter-specific
+    functions is not appropriate (i.e. not calculating tags for a CPython or
+    PyPy intrepreter).
+
+    The specific tags generated are:
+
+    - ``<interpreter>-<abi>-<platform>``
+    - ``<intepreter>-none-<platform>`` if ``"none"`` was not provided as part of
+      *abis*
+
+    :param str interpreter: The name of the interpreter. Defaults the current
+                            PyPy interpreter.
+    :param Iterable abis: Iterable of compatible ABIs. Defaults to the ABIs
+                          compatible with the current system.
+    :param Iterable platforms: Iterable of compatible platforms. Defaults to the
```

Comment:
It only works if there's no spaces in the name. So for instance this fails for `Tuple[int, int]`, and so I didn't bother adding type parameters when it will only work in some instances.

Reply:
Ah well, that makes sense.

I think we can visit how we're documenting these in a follow up issue/PR.